api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImpl.java

```diff
@@ -28,11 +28,16 @@
 	
 	@Inject
 	LocationService locationService;
-	
+
 	@Inject
 	@Named("sessionFactory")
 	SessionFactory sessionFactory;
-	
+
+	@Override
+	public Collection<Location> findAllLocations() {
+		return locationService.getAllLocations();
```

Comment:
Could we use the pattern `sessionFactory.getCurrentSession().createCriteria(...).list();` here?